### ISSUE TITLE
RW 0.33 uses Storybook version v6.1.21 while 0.32 used v6.2.9

### ISSUE THREAD

**Issue by realStandal** [redwoodjs/redwood#2747]
Running `yarn rw storybook` on a RW 0.32 project:

```
info @storybook/react v6.2.9
```

Running the same after updating to 0.33:

```
info @storybook/react v6.1.21
```

I didn't see any mention of Storybook in the update's [Breaking Changes](https://community.redwoodjs.com/t/redwood-v0-33/2176#breaking-9) or [Changelog](https://community.redwoodjs.com/t/redwood-v0-33/2176#changelog-4).

---

I'm not sure if this is the cause of Storybook not loading the `--static-dir`, resulting in a 404 for `mockServiceWorker.js`. There is also an issue I'm running into when using `'@storybook/addon-essentials'`, particularly the docs addon.

I haven't tried downgrading the essential addons to match Storybook, but it feels as if this downgrade was unintentional?

**Comment by thedavidprice** [comment]
@realStandal Hmm, this is indeed a strange result as we have not yet upgraded past Storybook v6.1.21:
- [Here is the dependency version](https://github.com/redwoodjs/redwood/blob/448780315c64f1dc90dbdd29e39729ce7de822b9/packages/core/package.json#L33) in Redwood v0.32
- And [here is the same dependency](https://github.com/redwoodjs/redwood/blob/d3e1b6ef26089a57decf6f2a25f57f10b1bf777e/packages/core/package.json#L33) in v0.33

I wonder if there was an error in the Storybook console output? Running `yarn storybook --version` would likely be a better way to determine this. You could also see if there are multiple versions installed with `yarn why @storybook/react`.

FYI: I'm working on upgrading Storybook as a part of this PR https://github.com/redwoodjs/redwood/pull/2402

> I'm not sure if this is the cause of Storybook not loading the --static-dir, resulting in a 404 for mockServiceWorker.js. There is also an issue I'm running into when using '@storybook/addon-essentials', particularly the docs addon.

Could you report these errors/problems with more detail in a separate Issue?

**Comment by realStandal** [comment]
Just for reference, here is the project I'm basing this issue off off: [`0.33`](https://github.com/LockTech/cerberus) | [`0.32`](https://github.com/LockTech/cerberus/tree/storybook-0.32); though just for looks, bit of effort to getting it standing.

> this is indeed a strange result as we have not yet upgraded past Storybook v6.1.21:

That makes me think I pulled Storybook in at `v6.2.9` when I installed `@storybook/addon-essentials`, however `yarn why @storybook/react` within the `v0.32` Redwood app produced the following:

```
❯ yarn why @storybook/react
yarn why v1.22.10
[1/4] Why do we have the module "@storybook/react"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "@storybook/react@6.2.9"
info Reasons this module exists
   - "_project_#@redwoodjs#core" depends on it
   - Hoisted from "_project_#@redwoodjs#core#@storybook#react"
info Disk size without dependencies: "396KB"
info Disk size with unique dependencies: "27.14MB"
info Disk size with transitive dependencies: "396KB"
info Number of shared dependencies: 434
```

Versus the `0.33` counterpart:

```
❯ yarn why @storybook/react
yarn why v1.22.10
[1/4] Why do we have the module "@storybook/react"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "@storybook/react@6.1.21"
info Reasons this module exists
   - "_project_#@redwoodjs#core" depends on it
   - Hoisted from "_project_#@redwoodjs#core#@storybook#react"
info Disk size without dependencies: "7.12MB"
info Disk size with unique dependencies: "7.12MB"
info Disk size with transitive dependencies: "7.12MB"
info Number of shared dependencies: 517
```

Same results save the difference in version and size.

---

* [Here is the `0.32` `yarn rw storybook` output.](https://gist.github.com/realStandal/cf6a53e74e99f83845d64a768cc3a759)
* [Here is `0.33`](https://gist.github.com/realStandal/51918c572c6fd31c7c8fb2147a0d8087)

Running `start-storybook` with the `--version` flag in `0.32`

```
❯ .\node_modules\.bin\start-storybook --version
6.2.9
```

Running it in `0.33`

```
❯ .\node_modules\.bin\start-storybook --version
6.1.21
```

**Comment by realStandal** [comment]
> Could you report these errors/problems with more detail in a separate Issue?

I certainly can and will!

**Comment by thedavidprice** [comment]
Wowza... I bet it's exactly what was happening here 🤔

https://github.com/redwoodjs/redwood/issues/2763#event-4859176131

**Comment by thedavidprice** [comment]
My advice to you is the same as I gave in https://github.com/redwoodjs/redwood/issues/2763#event-4859176131 — you _can_ manually add the `@storybook/react` dependency to your project at the latest version. You'll then want to monitor the version included in Redwood to remove the manually added dependency once it's behind (or else your project would then be using an older version).

I'm working on upgrading Storybook to the up-and-coming 6.3 (which supports Webpack 5) in this PR https://github.com/redwoodjs/redwood/pull/2402 if you want to follow along. Hoping to have it in within a release or two.

I'll leave it up to you to close this Issue if that's good for step 1. And I'll keep a look out for other Issues with the problems you mentioned above.

Sound good?

**Comment by realStandal** [comment]
Sounds good, thanks as always for the help @thedavidprice!

**Comment by realStandal** [comment]
I've *finally* gotten around to working on my web-side again, and just wanted to add a reference for how I've gotten around this issue.

First, I've installed `@storybook/react` and (for my needs) `@storybook/addon-essentials` at version `6.3.1`, done on the web-side of my application:

```
$ cd ./web
$ yarn add -D @storybook/react@6.3.1 @storybook/addon-essentials@6.3.1
```

Then, I've added a `resolutions` field to my projects **primary** `package.json`, located at the root of my application:

```YAML
{
  // ...
  "resolutions": {
    "@storybook/addon-essentials": "6.3.1",
    "@storybook/react": "6.3.1"
  }
}
```

The `resolutions` is added to circumnavigate Redwood's dependency on Storybook, which overrides my web-side's dependency; making version `6.1.X` the one which is actually installed and used.

---

I know I'm *incredibly* late for posting those other issues (`static-dir` and the docs error from `addon-essentials`), but installing the updated version of Storybook removed both of them.

[Here is the updated output from running `yarn rw storybook` after updating](https://gist.github.com/realStandal/8eb01f6f1071d80d2845a3fdd76ea6a8).
